testdata/project-v3-addon/.gitignore

```diff
@@ -6,6 +6,7 @@
 *.so
 *.dylib
 bin
+testbin/*
```

Comment:
Didn't see any changes in `pkg/plugins/golang/v3/scaffolds/templates/gitignore.go`.

Reply:
Self-reply: That line was added several months ago by #1626 and were never updated in the corresponding `testdata` directory. I think we should remove them from this PR and track this error separately, as it may mean that checking the `testdata` directory may be bugged.

Reply:
What is missing here is to update the Kubebuiler .gitignore to ensure that the bins and go.sum will not be committed for the new testdata sample see:
- https://github.com/kubernetes-sigs/kubebuilder/blob/master/.gitignore#L19
- https://github.com/kubernetes-sigs/kubebuilder/blob/master/.gitignore#L37